### PR TITLE
allow stats API to be disabled on a per HA basis. Improve test coverage

### LINKED ISSUE #1267: allow for stats API disabled on a per HA ID
/assign
/kind enhancement

### PR DESCRIPTION
**Pull request by mikehelmick**
Fixes #1267

## Proposed Changes

* Allow stats API to be disabled for a specific HA (enabled by default)
* Improve test coverage
* Fix make file, prow doesn't have awk apparently

**Release Note**

```release-note
Allow stats API to be enabled/disabled on a per HA basis.
```

### CODE REVIEW

File: internal/verification/authenticate_stats.go [hunk 1]
```diff
@@ -77,6 +77,11 @@ func (v *Verifier) AuthenticateStatsToken(ctx context.Context, rawToken string)
 		if !ok {
 			return nil, fmt.Errorf("incorrect type in cache: %w", err)
 		}
+		// check that the API is enabled for this HA.
+		if !healthAuthority.EnableStatsAPI {
```
Comment (sethvargo) [inline internal/verification/authenticate_stats.go:81]: We'll need to make a note to come back to this during productionization. This is a user error and should be 4xx not 5xx.

File: migrations/000055_AddDisableHAStats.down.sql [hunk 2]
```diff
@@ -0,0 +1,20 @@
+-- Copyright 2020 Google LLC
```
Comment (sethvargo) [inline migrations/000055_AddDisableHAStats.down.sql:1]: 2021

File: migrations/000055_AddDisableHAStats.up.sql [hunk 3]
```diff
@@ -0,0 +1,23 @@
+-- Copyright 2020 Google LLC
```
Comment (sethvargo) [inline migrations/000055_AddDisableHAStats.up.sql:1]: 2021

File: tools/admin-console/templates/healthauthority.html [hunk 4]
```diff
@@ -55,6 +55,17 @@
         </small>
       </div>
 
+      <div class="form-group">
+        <label for="enable-stats-api">Enable Stats API Access</label>
+        <select name="enable-stats-api" id="enable-stats-api" class="form-control custom-select">
```
Comment (sethvargo) [inline tools/admin-console/templates/healthauthority.html:60]: This is probably better served as a checkbox, but meh.

**Review by sethvargo — COMMENTED** [review]
/lgtm
/hold

File: internal/verification/authenticate_stats.go (hunk 1 above)
Comment (mikehelmick) [reply internal/verification/authenticate_stats.go:81]: this results in http.StatusUnauthorized

File: migrations/000055_AddDisableHAStats.down.sql (hunk 2 above)
Comment (mikehelmick) [reply migrations/000055_AddDisableHAStats.down.sql:1]: also updated template...

File: tools/admin-console/templates/healthauthority.html (hunk 4 above)
Comment (mikehelmick) [reply tools/admin-console/templates/healthauthority.html:60]: we use select a lot for this...

**Pull request merged**